src/main/scala/rps/service/GameService.scala

```diff
@@ -0,0 +1,26 @@
+package rps.service
+
+import scala.util.Random
+
+import rps.model.{GameSummary, Move, Result}
+import Move._, Result._
+import rps.model.error._
+import rps.repository.{GameRepository}
+import GameRepository._
+
+object GameService {
```

Comment:
Here you're not separating `GameService` from its interface, so it won't be possible (for instance) to test it in isolation.

Try replicating the same structure of `GameApi` (interface `trait` + implementation `class`)

Reply:
I would love if you could expand a little on this (but only if it doesn't bother you 😄)

Reply:
By having `GameService` to be an object, other components will depend on it directly, making it hard to mock its implementation. For example, if you want to test `GameApi` in isolation, you would need some magic library for mocking `GameService` out. If you instead make `GameApi` depend on a `GameService` trait, you can just implement a mocked version of it.